modules/savedb.py

```diff
@@ -22,41 +21,30 @@ def saveToDatabase(database, user, password, links):
 
         #Debug
         #print("Database:", database, "\nuser:",user, "\npass:",password)
-        db = MySQLdb.connect(host="localhost",    # your host
-                                user=user,         # your username
-                                passwd=password,  # your password
-                                db=database)        # name of the data base
-
-        cur = db.cursor()
-
         try:
-                query = """ CREATE TABLE IF NOT EXISTS `tor_url` (
-                id INT(6) UNSIGNED AUTO_INCREMENT PRIMARY KEY,
-                link VARCHAR(30) NOT NULL UNIQUE,
-                reg_date TIMESTAMP)"""
-                cur.execute(query)
-                for link in links:
-                        query = "INSERT IGNORE INTO `tor_url` (link) VALUES ('{0}')".format(link)
-                        cur.execute(query)
-                        #print(query)            
-                db.commit()
-                
-        except (MySQLdb.Error, MySQLdb.Warning) as e:
-                print(e)
-                return None
-
-        except TypeError as e:
-                print(e)
-                return None
-
-        except ValueError as e:
-                print(e)
-                return None
-
-        finally:
-                cur.close()
-                db.close()
-
-
-
-    
\ No newline at end of file
+            db = MySQLdb.connect(host="localhost1",    # your host
+                                    user=user,         # your username
+                                    passwd=password,  # your password
+                                    db=database)        # name of the data base
+
+            cur = db.cursor()
+            try:
+                    query = """ CREATE TABLE IF NOT EXISTS `tor_url` (
+                    id INT(6) UNSIGNED AUTO_INCREMENT PRIMARY KEY,
+                    link VARCHAR(30) NOT NULL UNIQUE,
+                    reg_date TIMESTAMP)"""
+                    cur.execute(query)
+                    for link in links:
+                            query = "INSERT IGNORE INTO `tor_url` (link) VALUES ('{0}')".format(link)
+                            cur.execute(query)
+                            #print(query)
+                    db.commit()
+
+            except (MySQLdb.Error, MySQLdb.Warning, TypeError, ValueError) as e:
+                    print(e)
+                    return None
+            finally:
+                    cur.close()
+                    db.close()
+        except:
```

Comment:
Why do you have a blank except right here? You can just use one try-except block instead of a nested one.

Reply:
The nested try is to separate errors. You can get an error for not being able to connect to database, or get an error for not being able to insert data into the table

Reply:
Yes there should be two completely separate try excepts. for example:
```
try:
    db = MySQLdb.connect(...)
    cur = db.cursor()
except:
    print("Unable to connect to database")

try:
    query = ...
except (MySQLdb.Error, MySQLdb.Warning, TypeError, ValueError) as e:
   print(e)```

Reply:
Thanks... Nice tip, didn't thought of it

Reply:
No problem, I just thought it'd make sense to separate the connection logic from the actual querying logic. Plus if you want blank exceptions to cover as little code as possible since it can be triggered by any error.